apps/web/src/components/dashboard/energy-aggregation-option.tsx

```diff
@@ -15,17 +17,49 @@ export default function EnergyAggregation({ selected }: Props) {
     const pathname = usePathname();
     const searchParams = useSearchParams();
 
+    const availableOptions = useMemo(() => {
+        const startDateStr = searchParams.get('start');
+        const endDateStr = searchParams.get('end');
+        if (!startDateStr || !endDateStr) {
+            return [AggregationType.RAW];
+        }
+
+        const startDate = new Date(startDateStr);
```

Comment:
Ab hier ist die Funktion aus [user-consumption-aggregation-option.tsx](https://github.com/pgenergy/energyleaf/pull/79/files#diff-f26eec090882e50e208011eadbf9212103edceb09230c402b303a36265919089) direkt kopiert. Am besten solltest du das in eine Methode auslagern